src/importlinter/contracts/layers.py

```diff
@@ -16,27 +16,30 @@
 
 
 @dataclass(frozen=True)
-class Layer:
+class ModuleTail:
     name: str
     is_optional: bool = False
 
 
+@dataclass(frozen=True)
+class Layer:
+    module_tails: set[ModuleTail]
```

Comment:
🤔 Might be worth making this a frozenset, so `Layer` becomes hashable - that allows us to add it to sets. On the other hand it's a bit more awkward to notate frozensets so I'll leave it up to you, we can always adjust later.

Reply:
I've left it as `set` for now, as you say it's pretty easy to adjust later if needed